projects/plugins/boost/app/assets/src/js/features/critical-css/critical-css-meta/critical-css-meta.tsx

```diff
@@ -47,32 +43,34 @@ const CriticalCssMeta: React.FC< CriticalCssMetaProps > = ( {
 						'jetpack-boost'
 					) }
 				</div>
-				<ProgressBar progress={ criticalCssProgress } />
+				<ProgressBar progress={ progress || 0 } />
 			</div>
 		);
-	} else if ( isFatalError ) {
-		return (
-			<ShowStopperError
-				status={ cssState.status }
-				primaryErrorSet={ primaryErrorSet }
-				statusError={ cssState.status_error }
-				regenerateCriticalCss={ retry }
-				showRetry={ ! hasRetried }
-			/>
-		);
+	} else if ( isFatalError( cssState ) ) {
+		return <ShowStopperError cssState={ cssState } retry={ retry } showRetry={ ! hasRetried } />;
 	}
 
 	return (
-		<Status
-			isCloudCssAvailable={ isCloudCssAvailable }
-			status={ cssState.status }
-			successCount={ successCount }
-			updated={ cssState.updated }
-			issues={ issues }
-			progress={ criticalCssProgress }
-			suggestRegenerate={ suggestRegenerate }
-		/>
+		<>
+			<Status
+				isCloudCssAvailable={ isCloudCssAvailable }
+				status={ cssState.status }
+				issues={ getCriticalCssIssues( cssState ) }
+				successCount={ successCount }
+				updated={ cssState.updated }
+				progress={ progress || 0 }
+				showRegenerateButton={ !! regenerateReason }
+			/>
+
+			<RegenerateCriticalCssSuggestion regenerateReason={ regenerateReason } />
+		</>
 	);
 };
 
-export default CriticalCssMeta;
+export default function ( props: CriticalCssMetaProps ) {
+	return (
+		<DataSyncProvider>
+			<CriticalCssMeta { ...props } />
+		</DataSyncProvider>
```

Comment:
We can skip wrapping in `<DataSyncProvider />` now since the `Main` component is wrapped with it.

Reply:
Fixed in ea856b911d.